include/deal.II/base/thread_management.h

```diff
@@ -692,16 +692,18 @@ namespace Threads
     private:
       RT value;
     public:
+      typedef RT &reference_type;
+
       inline return_value () : value() {}
 
-      inline RT get () const
+      inline reference_type get ()
       {
         return value;
       }
 
-      inline void set (RT v)
+      inline void set (RT &&v)
       {
-        value = v;
+        value = std::move(v);
```

Comment:
This is the key change of it all. It makes sure that whatever the function called on a separate thread/task returns, is moved, not copied, to the location where we store the result of the function so that the place that created the thread/task can access it.

Reply:
Is the call to `std::move` necessary? `v` is already an rvalue reference at that point.